hikari/internal/protocol.py

```diff
@@ -102,6 +105,29 @@ class Protocol(typing.Generic, metaclass=_FastProtocol):
 
     __slots__: typing.Sequence[str] = ()
 
+    @typing._tp_cache
+    def __class_getitem__(cls, params):
+        if not isinstance(params, tuple):
+            params = (params,)
+
+        if not params:
+            raise TypeError(f"Parameter list to {cls.__qualname__}[...] cannot be empty")
+
+        msg = "Parameters to generic types must be types."
+        params = tuple(typing._type_check(p, msg) for p in params)
+
+        if cls is Protocol:
+            # Generic and Protocol can only be subscripted with unique type variables.
+            if not all(isinstance(p, typing.TypeVar) for p in params):
+                raise TypeError("Parameters to Protocol[...] must all be type variables")
+            if len(set(params)) != len(params):
+                raise TypeError("Parameters to Protocol[...] must all be unique")
+        else:
+            # Subscripting a regular Generic subclass.
+            typing._check_generic(cls, params)
+
+        return typing._GenericAlias(cls, params)
```

Comment:
Are you actually ok with this fix and how it's become a pretty complex, edge casey, possibly difficult to maintain and possibly still broken in ways which we can't forsee "fix" which seems to just bring in even more issues and ends up having to rely the protected impl detail of stdlib stuffe for someting that could've otherwise been fixed by adding a few more attributes to places and also makes our "protocols" incompatible with standard protocols?

Reply:
As I said in the issue, i dont really like this, but it is the only way to fix it. I have been trying to figure out a better way, but none for the moment.

Reply:
Well as it stands this is probably one of the more breaking and complex approaches to fixing the issue of slow instance checks being used internally that could've been picked outside of just removing stuff as it involves changing pre-established parts of the public interface as opposed to just using a helper function which handles it using a different approach (maybe duck typing) or adding attributes which is an addition not a change and works better with static typing